src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/ColorTransform.cs

```diff
@@ -204,11 +181,9 @@ private long ICM2Color(float[] srcValue)
             }
             else
             {
-                byte[] channel = new byte[8];
+                Span<byte> channel = stackalloc byte[8];
 
```

Comment:
I think we should call clear here, just to make sure all values are initialized to zero.
channel.Clear();

Reply:
This one seems like a valid point, since the `srcValue.Length` can be less than 8, and all the 8 values are then used below.

Reply:
@hiteshwpfmsft @miloush I do not see the reason for that, same in the other case. If .localsinit is emitted, which is by default unless [SkipLocalsInit] attribute is specified (it is not here), all stackalloc memory is zero-init by default (besides that one-time when Roslyn didn't do it, which was a bug and was fixed).

Reply:
Hm are you suggesting this documentation is outdated?

> The content of the newly allocated memory is undefined.

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/stackalloc

Reply:
Very technically speaking, the documentation is correct. The `stackalloc` keyword doesn't gurantee the zero-init of the "allocated" memory. However, it is guranteed that Roslyn will emit .`localsinit` if `stackalloc` (`localloc` in IL) is present and the `SkipLocalsInit` attribute has not been specified.

And with that, it is guranteed that JIT will generate efficient stack zero-init loop. Anything else is a bug.

Reply:
More relevant documentation here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/skip-localsinit

Thank you @h3xds1nz, I am convinced this gets zeroed.

Reply:
Thanks guys for the discussion.